src/stories/ActionList.stories.tsx

```diff
@@ -362,3 +362,59 @@ export function SizeStressTestingStory(): JSX.Element {
   )
 }
 SizeStressTestingStory.storyName = 'Size Stress Testing'
+
+type ReactRouterLikeLinkProps = {to: string; children: React.ReactNode}
+const ReactRouterLikeLink = forwardRef<HTMLAnchorElement, ReactRouterLikeLinkProps>(
+  ({to, ...props}: {to: string; children: React.ReactNode}, ref) => {
+    // eslint-disable-next-line jsx-a11y/anchor-has-content
+    return <a ref={ref} href={to} {...props} />
+  }
+)
+
+const NextJSLikeLink = forwardRef(
+  ({href, children}: {href: string; children: React.ReactNode}, ref): React.ReactElement => {
+    const child = React.Children.only(children)
+    const childProps = {
+      ref,
+      href
+    }
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return React.cloneElement(child, childProps)
+  }
+)
+
+export function LinkItemStory(): JSX.Element {
+  return (
+    <>
+      <h1>Simple List</h1>
+      <ErsatzOverlay>
+        <ActionList
+          items={[
+            {
+              text: 'A. Vanilla action',
+              renderItem: props => <ActionList.Item onAction={() => alert('hi?')} {...props} />
+            },
+            {
+              text: 'B. Vanilla link',
+              renderItem: props => <ActionList.Item as="a" href="/about" {...props} />
+            },
+            {
+              text: 'C. React Router link',
+              renderItem: props => <ActionList.Item as={ReactRouterLikeLink} to="/about" {...props} />
+            },
+            {
+              text: 'D. NextJS style',
+              renderItem: props => (
+                <NextJSLikeLink href="/about">
+                  <ActionList.Item as="a" {...props} />
+                </NextJSLikeLink>
+              )
+            }
+          ]}
+        />
```

Comment:
Could we add a Jest test that will fail CI if `ActionList.Item` doesn't render the element passed through the `as` prop correctly?

I want to make sure we don't accidentally break this if we refactor `ActionList.Item` (which will likely happen soon)

Reply:
Just added a `behavesAsComponent` test